hal/armv7m/imxrt/timer.c

```diff
@@ -75,14 +115,38 @@ int hal_timerRegister(int (*f)(unsigned int, cpu_context_t *, void *), void *dat
 
 void _hal_timerInit(u32 interval)
 {
-	timer_common.jiffies = 0;
+	timer_common.base = (void *)GPT_BASE;
 
-	_imxrt_systickInit(interval);
+	/* Disable timer */
+	*(timer_common.base + gpt_cr) &= ~1;
```

Comment:
because of use of a signed integer operand with a unary bitwise operator
```suggestion
	*(timer_common.base + gpt_cr) &= ~1U;
```

Reply:
Not strictly needed as numeric value of the result this operation is undefined indeed, but bit pattern we care about here is well defined. Changed anyway